qdax/core/mels.py

```diff
@@ -0,0 +1,120 @@
+"""Core components of the MAP-Elites Low-Spread algorithm."""
+from __future__ import annotations
+
+from functools import partial
+from typing import Callable, Optional, Tuple
+
+import jax
+import jax.numpy as jnp
+
+from qdax.core.containers.mels_repertoire import MELSRepertoire
+from qdax.core.emitters.emitter import Emitter, EmitterState
+from qdax.core.map_elites import MAPElites
+from qdax.types import (
+    Centroid,
+    Descriptor,
+    ExtraScores,
+    Fitness,
+    Genotype,
+    Metrics,
+    RNGKey,
+)
+
+
+class MELS(MAPElites):
+    """Core elements of the MAP-Elites Low-Spread algorithm.
+
+    Most methods in this class are inherited from MAPElites.
+
+    The same scoring function can be passed into both MAPElites and this class.
+    We have overridden __init__ such that it takes in the scoring function and
+    wraps it such that every solution is evaluated `num_evals` times.
+
+    We also overrode the init method to use the MELSRepertoire instead of
+    MapElitesRepertoire.
+    """
+
+    def __init__(
+        self,
+        scoring_function: Callable[
+            [Genotype, RNGKey], Tuple[Fitness, Descriptor, ExtraScores, RNGKey]
+        ],
+        emitter: Emitter,
+        metrics_function: Callable[[MELSRepertoire], Metrics],
+        num_evals: int,
+    ) -> None:
+        def multi_eval_scoring_function(
```

Comment:
Minor suggestion: put this function on top of the file with a `make_multi_eval_scoring_function` to reduce nested indents.

Reply:
Agreed